### ISSUE TITLE
DICe execution failed

### ISSUE THREAD

**Issue by Bestzi** [dicengine/dice#315]
![error dice](https://github.com/dicengine/dice/assets/159171321/605a8efa-fecd-4eae-a8ea-61f317673d1a)
image dimensions: 2448 x 2048
Output will be written to one file per frame with all subsets included
Execution information will be written to a separate file (not placed in the output headers)
C:\Users\dzturne\Software\DICe_nightly\src\core\DICe_Decomp.cpp:73:

Throw number = 1

Throw test that evaluated to true: comm_->get_rank()==0&&subset_centroids_x.size()<=0
//////
Excuse me, what the problem is about ?

**Comment by Bestzi** [comment]
![error 2 dice](https://github.com/dicengine/dice/assets/159171321/6f5c081d-8949-468b-ab46-7a4972a6c48e)

How about this one. Both images are from my camera and camera resolution is 2448x2048. You
Thank you in advance.

**Comment by dicengine** [comment]
For the first case above, it's hard to say what's going wrong since I can't see all the parameters you have selected on the right panel. One possibility is that there is that the SSSIG threshold is too high. Have you tried setting that to zero to see if it works? There might not be enough contrast in the top image to satisfy the SSSIG criteria. Also, if you could attach the input.xml file that got generated in the working directory I could help you more.

For the bottom case, the problem is that you selected "calibrate results" but you did not provide the calibration parameters in the "load cal file" input box. Did you calibrate the camera using the perform calibration tool? If not you won't be able to calibrate the resutls.

**Comment by Bestzi** [comment]
Thank you for your help for a last problem.
There's another problem that I quiet not understand about it. 
![error 3 dice](https://github.com/dicengine/dice/assets/159171321/4e5d445d-4c58-4b8e-8465-0a9b0b9bba21)

Thank you in advance.

**Comment by dicengine** [comment]
If you chose to install DICe in a folder that is not the default location suggested by the install dialogue you have to select the folder where the dice.exe executable exists. The reason you get an error about DICe_OpenCVServer.exe is because that is the first executable that DICe looks for when you open the GUI. You can either reinstall DICe in the default location or re-install it and when you get the first error above about the executable path not existing, select the folder where you have dice.exe installed in the dialog box.

**Comment by Bestzi** [comment]
Well, I can run properly now thank you a lot for your information. However, I have my own calibration code, but I'm wondering what is in the .xml file. What does it look like inside the file? How can I see what is written in the .xml file?

Thank you in advance.

**Comment by Bestzi** [comment]
I have to tell you that I can't calibrate like in your example clip. I don't understand why but I really just want to know what's inside the .xml file and what does it looks like.

**Comment by dicengine** [comment]
Here's an example of the DICe xml file: https://github.com/dicengine/dice/blob/master/tests/regression/stereo_rectified/cal.xml

**Comment by Bestzi** [comment]
If I have key point value which could I insert the value to let the program calibrate ?

**Comment by dicengine** [comment]
We don't have the ability in DICe to read in the keypoint locations and then calibrate, but you could use OpenCV to write a small python script that does that. Once you calibrate using OpenCV you can just copy and paste the intrinsic and extrinsic parameters to a DICe formated cal.xml file or have the python script output the cal.xml file.

**Comment by Bestzi** [comment]
Thank you, could you tell the problem about this right picture I've tried to use another picture but it still didn't worked ?   
![image](https://github.com/dicengine/dice/assets/159171321/7d17da89-fee6-4305-8170-af5e50471375)

**Comment by Bestzi** [comment]
> We don't have the ability in DICe to read in the keypoint locations and then calibrate, but you could use OpenCV to write a small python script that does that. Once you calibrate using OpenCV you can just copy and paste the intrinsic and extrinsic parameters to a DICe formated cal.xml file or have the python script output the cal.xml file.
Sorry but I'm hard understanding so this is my intrinsic parameters from python code.
![image](https://github.com/dicengine/dice/assets/159171321/ba565f7f-74cc-4931-9250-562e89deab00)
by [ cx cy fx fs fy k1 k2 k3] and I want to use my parameter instead of DICe but I have to use your form of cal.xml so could you give me and illustrate some of examples and how? many thank you from me.

**Comment by dicengine** [comment]
That's right, you can simply change the values in your cal.xml file if you have a template from DICe to work from.

**Comment by dicengine** [comment]
The problem you are seeing when you try to run the DICe calibration on the images is probably due to the brightness changes across your calibration target. It makes setting the segmentation threshold hard to get right. It just so happens that we recently updated the calibration routine to handle this better, but haven't pushed the changes yet. These updates will be available in a new version of DICe in a few weeks.

**Comment by Bestzi** [comment]
Thank you, about the previous question. Could you explain about these parameter ? 
![image](https://github.com/dicengine/dice/assets/159171321/53d658da-97f2-4af7-80d8-dee1f68c0822)

And also about the Euler angles.
![image](https://github.com/dicengine/dice/assets/159171321/75ed036c-2426-4201-88bb-05a8815ca53b)

Regards,
Best

**Comment by dicengine** [comment]
All of this is covered here: https://github.com/dicengine/dice/blob/master/doc/reports/Triangulation.pdf 
and here: https://docs.opencv.org/4.x/dc/dbb/tutorial_py_calibration.html
Note that under the hood, the DICe calibration tool uses the OpenCV calibration routines.

**Comment by Bestzi** [comment]
Thank you, but I have to bother you again so how do you calculated the Euler angles? I just wonder how the process looks like.

**Comment by dicengine** [comment]
Here's a link that might be helpful: https://learnopencv.com/rotation-matrix-to-euler-angles/

**Comment by Bezutoz** [comment]
Thank you for always helping me. I'm a student who doesn't understand much when reading. It might be asking too much. Anyway, thank you. Firstly, I still have a question about skew parameters (fs) because I don't see them in the cal.xml file: My camera has fs value so I want to input it but how do I do it and if I input will there be any problems?

Secondly, I would like to know where the openCV calibration file is and I want to add my keypoints to the process.

And the extrinsic parameters. Is it CAL_EXTRINSICS ? So it already evaluate between camera1 and 2  ? and about the Euler angles it doesn't need for evaluation because we already had a rotation matrix. Am I correct ?
![IMG_2919](https://github.com/dicengine/dice/assets/115972030/fd712317-c851-4568-b417-20f9b8438043)

Regards,
Best

**Comment by Bestzi** [comment]
I'm sorry, I meant that the rotation matrix value in the cal.xml file must be for camera 0 and camera 1, but for camera 0 it is a constant value. Does that mean that for Camera 1 it was calculated as what I highlighted?
![4 meeting](https://github.com/dicengine/dice/assets/159171321/6a10c763-f741-4055-9a80-439e98d95eb6)

**Comment by dicengine** [comment]
Yes, that's correct.

**Comment by Bezutoz** [comment]
Thank you, from your previous file Calibration.cpp you also have the .exe file to call the function of .cpp or where do you call and use the function ?

Regards,

**Comment by Bezutoz** [comment]
How can I output the function 'intersection_points'
because I want to compare with my data .

**Comment by Bestzi** [comment]
I actually want to know about "TrilinosConfig.cmake" file and how can I find this file? 

Regards,
